packages/url_launcher/url_launcher_web/example/integration_test/link_widget_test.dart

```diff
@@ -972,6 +971,109 @@ void main() {
+              return GestureDetector(
+                onTap: () {},
+                child: const Text('My Link'),
+              );
+            },
+          ),
+        ),
+      ));
+      // Platform view creation happens asynchronously.
+      await tester.pumpAndSettle();
+      await tester.pump();
+
+      final html.Element semanticsHost =
+          html.document.createElement('flt-semantics-host');
+      html.document.body!.append(semanticsHost);
+      final html.Element semanticsAnchor = html.document.createElement('a')
+        ..setAttribute('id', 'flt-semantic-node-99')
+        ..setAttribute('flt-semantics-identifier', 'test-link-71')
+        ..setAttribute('href', uri.toString())
+        ..textContent = 'My Text Link';
+      semanticsHost.append(semanticsAnchor);
+
+      expect(pushedRouteNames, isEmpty);
+      expect(testPlugin.launches, isEmpty);
+
+      // Receive the DOM click first.
+      final html.Event event = _simulateClick(semanticsAnchor);
+      // The browser will be prevented from navigating right at the end of the event loop because
+      // we are still not sure if the `followLink` will arrive or not.
+      await Future<void>.delayed(Duration.zero);
+      expect(event.defaultPrevented, isTrue);
+      // After some delay, the engine's click debouncer sends the events to the framework and we get
+      // the `followLink` signal.
+      await Future<void>.delayed(const Duration(milliseconds: 200));
+      await followLinkCallback!();
+
+      expect(pushedRouteNames, isEmpty);
+      expect(testPlugin.launches, <String>['https://flutter.dev']);
+    });
+
+    // Regression test for: https://github.com/flutter/flutter/issues/162927
+    testWidgets('prevents navigation with debounced clicks on semantic link',
+        (WidgetTester tester) async {
+      final Uri uri = Uri.parse('https://flutter.dev');
+
+      await tester.pumpWidget(MaterialApp(
+        home: WebLinkDelegate(
+          semanticsIdentifier: 'test-link-71',
+          TestLinkInfo(
+            uri: uri,
+            target: LinkTarget.blank,
+            builder: (BuildContext context, FollowLink? followLink) {
+              return GestureDetector(
+                onTap: () {},
+                child: const Text('My Link'),
+              );
+            },
+          ),
+        ),
+      ));
+      // Platform view creation happens asynchronously.
+      await tester.pumpAndSettle();
+      await tester.pump();
+
+      final html.Element semanticsHost =
+          html.document.createElement('flt-semantics-host');
+      html.document.body!.append(semanticsHost);
+      final html.Element semanticsAnchor = html.document.createElement('a')
+        ..setAttribute('id', 'flt-semantic-node-99')
+        ..setAttribute('flt-semantics-identifier', 'test-link-71')
+        ..setAttribute('href', uri.toString())
+        ..textContent = 'My Text Link';
+      semanticsHost.append(semanticsAnchor);
```

Comment:
Is this needed because we can't turn on semantics in the test, so the engine creates all the semantics nodes for the Link?

Reply:
Semantics mode doesn't work end-to-end in tests, so I have to mimic what the engine would've created.